mentoring/mentoring.py

```diff
@@ -25,70 +22,275 @@
+        'display_name', 'mode', 'followed_by', 'max_attempts', 'enforce_dependency',
+        'display_submit', 'weight',
+    )
     icon_class = 'problem'
+    has_score = True
+    has_children = True
+
+    block_settings_key = 'mentoring'
+    theme_key = 'theme'
+
+    def _(self, text):
+        """ translate text """
+        return self.runtime.service(self, "i18n").ugettext(text)
+
+    @property
+    def is_assessment(self):
+        """ Checks if mentoring XBlock is in assessment mode """
+        return self.mode == 'assessment'
+
+    def get_theme(self):
+        """
+        Gets theme settings from settings service. Falls back to default (LMS) theme
+        if settings service is not available, xblock theme settings are not set or does
+        contain mentoring theme settings.
+        """
+        settings_service = self.runtime.service(self, "settings")
+        if settings_service:
+            xblock_settings = settings_service.get_settings_bucket(self)
+            if xblock_settings and self.theme_key in xblock_settings:
+                return xblock_settings[self.theme_key]
+        return _default_theme_config
+
+    @property
+    def score(self):
+        """Compute the student score taking into account the weight of each step."""
+        weights = (float(self.runtime.get_block(step_id).weight) for step_id in self.steps)
+        total_child_weight = sum(weights)
+        if total_child_weight == 0:
+            return Score(0, 0, 0, 0, 0)
+        score = sum(r[1]['score'] * r[1]['weight'] for r in self.student_results) / total_child_weight
+        correct = sum(1 for r in self.student_results if r[1]['status'] == 'correct')
+        incorrect = sum(1 for r in self.student_results if r[1]['status'] == 'incorrect')
+        partially_correct = sum(1 for r in self.student_results if r[1]['status'] == 'partial')
+
+        return Score(score, int(round(score * 100)), correct, incorrect, partially_correct)
+
+    def include_theme_files(self, fragment):
+        theme = self.get_theme()
+        theme_package, theme_files = theme['package'], theme['locations']
+        for theme_file in theme_files:
+            fragment.add_css(ResourceLoader(theme_package).load_unicode(theme_file))
 
     def student_view(self, context):
-        fragment, named_children = self.get_children_fragment(context, view_name='mentoring_view',
-                                                              not_instance_of=MentoringMessageBlock)
+        # Migrate stored data if necessary
+        self.migrate_fields()
 
-        correct_icon_url = self.runtime.local_resource_url(self, 'public/img/correct-icon.png')
+        fragment = Fragment()
+        child_content = u""
 
-        fragment.add_content(render_template('templates/html/mentoring.html', {
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if isinstance(child, MentoringMessageBlock):
+                pass
```

Comment:
Is this the right logic? If so, why not use `if not isinstance(..):` and leave out the pass and the else?